### ISSUE TITLE
Require authentication

### ISSUE THREAD

**Issue by boredland** [coinbase/mongobetween#68]
I managed to run mongobetween between my mongodb replica and me, but wonder how if there's a way to secure that connection with a username/password.

As I understand the `-username` and `-password` options are to be only applied to the downstream connection. Is there a way? I basically would want the same password to be required, that is used to connect to the replica.

**Comment by ThatHurleyGuy** [comment]
You're correct that the `-usename`/`-password` options are applied to the connection to the actual mongo cluster. There is not currently any mechanism built into mongobetween to provide authentication to connect to it. We run it as a sidecar, so it's only accessible on the physical host it's running on which negates the need to additional auth there. If you wanted to run it on a separate host, you'll need to either build that auth in at a higher level (locking it down at the network layer) or find a way to build the authentication into mongobetween itself. As for the latter, I don't think there's any easy mechanism to do that right now unfortunately

**Comment by boredland** [comment]
Do you think it is possible to add such a functionality to the proxy and would you accept a PR for that? I am missing any relevant GO-Skills, but perhaps someone could be found - so many people are stuck with MongoDB-Atlas & AWS Lambda for example - many of them could profit from a little container with MongoBetween running in.

**Comment by boredland** [comment]
Sadly we're currently stuck with Vercel, a cloud provider which to my knowledge wraps lambda, but surely does not allow access to anything close to running a container or defining a security group.

**Comment by boredland** [comment]
Closing this myself as the reporter — I'm no longer pursuing it, and I'd rather not leave it sitting in your triage queue.

No action is needed from the maintainers, and this says nothing about the validity of the report: if anyone else is hitting the same thing, please reopen or file fresh.

**Comment by boredland** [comment]
Reopening — please disregard my previous comment.

That closure was a mistake on my side: it came from a bulk sweep of my own filed issues, not from a considered decision to withdraw this report. Sorry for the noise in your notifications.

The report stands as originally written.
